src/Filter/ChoiceFilter.php

```diff
@@ -65,51 +88,62 @@ private function filterWithMultipleValues(ProxyQueryInterface $queryBuilder, str
         }
 
         $isNullSelected = \in_array(null, $data['value'], true);
-        $data['value'] = array_filter($data['value'], static function ($value): bool {
-            return null !== $value;
+        $data['value'] = array_filter($data['value'], static function ($data): bool {
+            return null !== $data;
         });
 
         // Have to pass IN array value as parameter. See: http://www.doctrine-project.org/jira/browse/DDC-3759
         $completeField = sprintf('%s.%s', $alias, $field);
-        $parameterName = $this->getNewParameterName($queryBuilder);
+        $parameterName = $this->getNewParameterName($query);
         if (EqualOperatorType::TYPE_NOT_EQUAL === $data['type']) {
-            $andConditions = [$queryBuilder->expr()->isNotNull($completeField)];
+            $andConditions = [$query->getQueryBuilder()->expr()->isNotNull($completeField)];
             if (0 !== \count($data['value'])) {
-                $andConditions[] = $queryBuilder->expr()->notIn($completeField, ':'.$parameterName);
-                $queryBuilder->setParameter($parameterName, $data['value']);
+                $andConditions[] = $query->getQueryBuilder()->expr()->notIn($completeField, ':'.$parameterName);
+                $query->getQueryBuilder()->setParameter($parameterName, $data['value']);
             }
-            $this->applyWhere($queryBuilder, $queryBuilder->expr()->andX()->addMultiple($andConditions));
+            $this->applyWhere($query, $query->getQueryBuilder()->expr()->andX()->addMultiple($andConditions));
         } else {
-            $orConditions = [$queryBuilder->expr()->in($completeField, ':'.$parameterName)];
+            $orConditions = [$query->getQueryBuilder()->expr()->in($completeField, ':'.$parameterName)];
             if ($isNullSelected) {
-                $orConditions[] = $queryBuilder->expr()->isNull($completeField);
+                $orConditions[] = $query->getQueryBuilder()->expr()->isNull($completeField);
             }
-            $this->applyWhere($queryBuilder, $queryBuilder->expr()->orX()->addMultiple($orConditions));
-            $queryBuilder->setParameter($parameterName, $data['value']);
+            $this->applyWhere($query, $query->getQueryBuilder()->expr()->orX()->addMultiple($orConditions));
+            $query->getQueryBuilder()->setParameter($parameterName, $data['value']);
         }
     }
 
-    private function filterWithSingleValue(ProxyQueryInterface $queryBuilder, string $alias, string $field, array $data = []): void
+    private function filterWithSingleValue(BaseProxyQueryInterface $query, string $alias, string $field, array $data = []): void
     {
+        /* NEXT_MAJOR: Remove this deprecation and update the typehint */
+        if (!$query instanceof ProxyQueryInterface) {
```

Comment:
It's not necessary to trigger a deprecation in this private method, just the `NEXT_MAJOR` comment.

Reply:
Indeed, I also discovered I also added `NEXT_MAJOR` comments for something I did on master https://github.com/sonata-project/SonataDoctrineORMAdminBundle/pull/1011